pipelines/skylab/optimus/scripts/parse_terra_tsv.py

```diff
@@ -29,34 +29,41 @@ def create_output_files(input_file,output_file,output_set):
     # for each fastq read, create a lane
     n_lanes = r1_fastq.shape[1] #number of fastq reads
     n_participants = r1_fastq.shape[0] #number of participants
-
-    column_names = ['entity:participant_lane_id', 'bundle_uuid', 'sample_id', 'r1_fastq','r2_fastq', 'i1_fastq']
+    column_names = ['entity:participant_lane_id', 'input_id', 'input_name','input_id_metadata_field','input_name_metadata_field', 'r1_fastq','r2_fastq', 'i1_fastq']
     participant_df = pd.DataFrame(columns = column_names)
 
     for j in range(n_participants):
         a  = []
         for i in range(n_lanes):
-            a.append("lane_"+str(i)+"_participant_"+str(j)+"_"+str(df.sample__provenance__document_id[j])+"_id")
+            a.append("participant_"+str(j)+"_lane_"+str(i)+"_"+str(df.sequencing_process__provenance__document_id[j])+"_id")
+    
         lane_id = pd.DataFrame({"entity:participant_lane_id":a})
         lane_fastq_r1 = pd.DataFrame({"fastq1":r1_fastq.iloc[j].to_numpy()})
         lane_fastq_r2 = pd.DataFrame({"fastq2":r2_fastq.iloc[j].to_numpy()})
         lane_fastq_i1 = pd.DataFrame({"fastqi":i1_fastq.iloc[j].to_numpy()})
-        bundle_uuid = pd.DataFrame({"entity:lane_id":np.repeat(df.bundle_uuid[j],n_lanes)})
-        sample_id = pd.DataFrame({"sample_id":np.repeat(df.sample__provenance__document_id[j],n_lanes)})
+        input_id = pd.DataFrame({"input_id":np.repeat(df.sequencing_process__provenance__document_id[j],n_lanes)})
+        input_id_metadata_field = pd.DataFrame({"input_id_metadata_field":np.repeat("sequencing_process.provenance.document_id",n_lanes)})
+        input_name = pd.DataFrame({"input_name_metadata_field":np.repeat(df.sequencing_input__biomaterial_core__biomaterial_id[j],n_lanes)})
+        input_name_metadata_field = pd.DataFrame({"entity:lane_id":np.repeat("sequencing_input.biomaterial_core.biomaterial_id",n_lanes)})
+
+        column_names = ['entity:participant_lane_id', 'input_id', 'input_name','input_id_metadata_field','input_name_metadata_field', 'r1_fastq','r2_fastq', 'i1_fastq']
```

Comment:
Does the order of the column name here matter (does it need to match the order they are assigned above)?

Reply:
I changed the script, the order has to match!